tests/fixtures/joint.py

```diff
@@ -43,15 +49,46 @@ class JointFixture:
     mongodb: MongoDbFixture
     rest_client: AsyncTestClient
 
+    def remove_db_data(self) -> None:
+        """Delete everything in the database to start from a clean slate"""
+        self.mongodb.empty_collections()
 
-@pytest_asyncio.fixture
-async def joint_fixture(
-    populated_mongodb_fixture: MongoDbFixture,  # noqa: F811
+    def load_test_data(self) -> None:
+        """Populate a collection for each file in test_data"""
+        filename_pattern = re.compile(r"/(\w+)\.json")
+        for filename in glob.glob("tests/fixtures/test_data/*.json"):
+            match_obj = re.search(filename_pattern, filename)
+            if match_obj:
+                collection_name = match_obj.groups()[0]
+                resources = get_resources_from_file(filename)
+                self.mongodb.client[self.config.db_name][collection_name].insert_many(
+                    resources
+                )
+                self.mongodb.client[self.config.db_name][collection_name].create_index(
+                    keys=[("$**", TEXT)]
+                )
+
+    async def call_search_endpoint(
+        self, search_parameters: JsonObject
+    ) -> models.QueryResults:
+        """Convenience function to call the /rpc/search endpoint"""
+        response = await self.rest_client.post(
+            url="/rpc/search", json=search_parameters
+        )
+        results = models.QueryResults(**response.json())
+        return results
+
+
+async def joint_fixture_function(
+    mongodb_fixture: MongoDbFixture,  # noqa: F811
 ) -> AsyncGenerator[JointFixture, None]:
-    """A fixture that embeds all other fixtures for API-level integration testing"""
+    """A fixture that embeds all other fixtures for API-level integration testing
+
+    **Do not call directly** Instead, use get_joint_fixture().
+    """
 
     # merge configs from different sources with the default one:
-    config = get_config(sources=[populated_mongodb_fixture.config])
+    config = get_config(sources=[mongodb_fixture.config])
 
     # create a DI container instance:translators
     async with get_configured_container(config=config) as container:
```

Comment:
Here right after the `async with` statement there is a `container.wire` statement that duplicates code from the main module. Therefore the test does not notice if in the main module the container is not wired properly. We had this problem in another service. The tests ran fine, but in deployment it failed because the container.wire in the main module was done incorrectly.

So I suggest moving the container.wire into the get_configured_container function.

Reply:
This is a pretty good point, maybe we should implement that change wherever it applies as PRs come up.
[6ec6114](https://github.com/ghga-de/mass/pull/9/commits/6ec6114680f8057fb87211b1a9bc466ffdb2a1cd)